src/modules/video_player/index.js

```diff
@@ -110,13 +110,62 @@ function togglePictureInPicture() {
     }
 }
 
+function insertPiPButton() {
+    const $anchor = $('.player-controls__right-control-group');
+    const $settingsButton = $anchor.children('div').children('div.settings-menu-button-component').parent();
+    const $button = $(getPictureInPictureTemplate(false));
+
+    $anchor.on('click', BTTV_PICTURE_IN_PICTURE_SELECTOR, togglePictureInPicture);
+    $button.insertAfter($settingsButton);
+}
+
+function loadPictureInPicture() {
+    if (!document.pictureInPictureEnabled || $(BTTV_PICTURE_IN_PICTURE_SELECTOR).length > 0) return;
+
+    const observe = (_watcher, element) => {
+        if (!element) return;
+        if (_watcher) _watcher.disconnect();
+        _watcher.observe(element, {
+            attributes: true,
+            attributeFilter: ['data-a-player-state'],
+            attributeOldValue: true
+        });
+    };
+
+    const miniPlayerObserver = new window.MutationObserver(mutations =>
+        mutations.forEach(mutation => {
+            if (mutation.oldValue === 'mini') {
+                insertPiPButton();
+            }
+        })
+    );
+
+    observe(miniPlayerObserver, $('.persistent-player')[0]);
```

Comment:
lets avoid using an observer for this. there's gotta be a better way to detect this behavior